jbcli/jbcli/utils/reload.py

```diff
@@ -0,0 +1,41 @@
+"""Handles watcher-related tasks for app reloading
+"""
+import os
+import time
+
+import click
+from requests import get
+
+
+def create_browser_instance():
+    """Create proxy browser instance for hot reloading
+    """
+    cmd = 'npx browser-sync start --proxy="localhost:8000"'
+    os.system(cmd)
+
+
+def refresh_browser(timeout=None):
```

Comment:
I have a nitpick / style suggestion for this function that I think would make the code simpler and shorter.


```python
while True: # or, maybe `for i in range(10)`, if you want to limit how long we'll wait
    if timeout:
        time.sleep(timeout)
    try:
        response = get('http://localhost:8000/health_check')
        if response and response.status_code == 200:
            click.echo('Refreshing browser...')
            cmd = "npx browser-sync reload"
            os.system(cmd)
            return
    except:
        click.echo('Almost done...')
```

This way we don't recurse (which might eventually blow the stack), and we don't need to define the inner `attempt_refresh` function.

Reply:
Great suggestion!